test/external/external_test_whisper_librispeech.py

```diff
@@ -0,0 +1,79 @@
+import unittest
+import torch
+import tqdm
+import torchaudio
+import pathlib
+import jiwer
+import os
+import numpy as np
+from whisper.normalizers import EnglishTextNormalizer
+from examples.whisper import init_whisper, transcribe_waveform
+
+class TestWhisperLibriSpeech(unittest.TestCase):
+  # target_wer determined by running https://github.com/openai/whisper/blob/main/notebooks/LibriSpeech.ipynb
+  # the values are consistent with the paper D.1.1 https://cdn.openai.com/papers/whisper.pdf#page=22
+  def test_en_tiny(self):
+    run_evaluation("tiny.en", 0.05655609406528749)
+
+  def test_tiny(self):
+    run_evaluation("tiny", 0.07558413638335187)
+
+  def test_en_base(self):
+    run_evaluation("base.en", 0.04271408904897505)
+
+  def test_en_small(self):
+    run_evaluation("small.en", 0.030531615969223228)
+
+def run_evaluation(model_name, target_wer):
+  dataset = LibriSpeech()
+  loader = torch.utils.data.DataLoader(dataset, batch_size=16)
+
+  model, enc = init_whisper(model_name)
+
+  hypotheses = []
+  references = []
+
+  for audio, texts in tqdm.tqdm(loader):
+    transcriptions = transcribe_waveform(model, enc, audio.numpy())
+    hypotheses.extend(transcriptions)
+    references.extend(texts)
+
+  normalizer = EnglishTextNormalizer()
+  normalized_hypotheses = [normalizer(text) for text in hypotheses]
+  normalized_references = [normalizer(text) for text in references]
+  wer = jiwer.wer(normalized_hypotheses, normalized_references)
+
+  np.testing.assert_almost_equal(wer, target_wer)
+
+class LibriSpeech(torch.utils.data.Dataset):
```

Comment:
Can this use tinygrad abstractions instead of torch ones?

Reply:
Which abstractions do you have in mind? We have https://github.com/tinygrad/tinygrad/blob/master/extra/datasets/librispeech.py which requires the user to manually download/process the dataset. I preferred using torch here because

1. It is exactly what the OpenAI eval is doing, increasing probability of matching WER
2. It automatically downloads and parses the dataset, providing the audio/text out of the box

Reply:
I'm fine with the torchaudio dataset, but why do you need `torch.utils.data.Dataset`?

Reply:
Oh, I thought I had to subclass `torch.utils.data.Dataset` in order to pass it to the `torch.utils.data.DataLoader`. Turns out I don't. Will remove in next revision.

Reply:
Oh, I see that's the return type of torchaudio anyway, ok this is fine.